### PR TITLE
transmission: use freeformType on settings and various fixes

### PR DESCRIPTION
**Pull request by ju1m**
###### Motivation for this change
Leveraging [freeformType](https://github.com/NixOS/nixpkgs/pull/82743) on `services.transmission.settings`.

###### Things done
- [X] Move settings assertions to options.
- [X] Alias option `port` to `settings.rpc-port`.
- [X] Alias option `openFirewall` to `openPeerPorts`.
- [X] Add option `openRPCPort`.
- [X] Add option `settings.trash-original-torrent-files`.
- [X] Add `mkDefault`s when setting `boot.kernel.sysctl`.
- [X] Use `<xref linkend="opt-services.transmission.${optionName}"/>` in `description`s.
- [X] Fix https://github.com/NixOS/nixpkgs/issues/135695 by removing the `InaccessiblePaths=` on the `RootDirectory=`'s path.
- [X] Remove `SystemCallErrorNumber=EPERM` which may silent failures to call a disallowed syscalls instead of failing hard with an `ESYS`, that will enable us to know more easily if we need to relax the `SystemCallFilter=` (using `coredumpctl debug` to know which syscall failed).
- [X] Fix `watch-dir` creation.

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by aanderse** [comment]
`openFirewall` is a pretty standard option name across modules. Do we want to leave it in as an alias?

File: nixos/modules/services/torrent/transmission.nix [hunk 1]
```diff
@@ -27,45 +32,140 @@ in
...
           See <link xlink:href="https://github.com/transmission/transmission/wiki/Editing-Configuration-Files">Transmission's Wiki</link>
-          for documentation.
+          for documentation of settings not explicitely covered by this module.
         '';
+        default = {};
+        type = types.submodule {
+          freeformType = with types;
+            (attrsOf (nullOr (oneOf [str int bool]))) // {
+              description = "setting option";
+            };
```
Comment (Infinisil) [inline nixos/modules/services/torrent/transmission.nix:49]: How about using `pkgs.formats.json` as introduced in https://github.com/NixOS/nixpkgs/pull/75584 instead. If you use its generator you'll also get a nicely formatted json file too.

**Comment by Infinisil** [comment]
@aanderse There's different options for different ports now, `openPeerPorts` and `openRPCPort`. I think it makes sense to not use `openFirewall` to avoid confusion.

**Comment by aanderse** [comment]
> @aanderse There's different options for different ports now, `openPeerPorts` and `openRPCPort`. I think it makes sense to not use `openFirewall` to avoid confusion.

I hadn't seen a PR adding that. Great news. I'll look into it :+1:

**Review by infinisil — CHANGES_REQUESTED** [review]
In addition to the things below, the descriptions of options `downloadDirPermissions` and `home` can be updated to reference the new options.

Also as another improvement to this module: There's `homeDir` defined in a `let in` at the top which can be removed (its usages need to be fixed though).

File: nixos/modules/services/torrent/transmission.nix (hunk 1 above)
Comment (ju1m) [reply nixos/modules/services/torrent/transmission.nix]: Nice, I've switched to that, thanks. Note that the formatting is done anyway by `jq` when adding secrets in `ExecStartPre=` and in`transmission-daemon` which overwrites its config.

File: nixos/modules/services/torrent/transmission.nix [hunk 2]
```diff
@@ -27,45 +32,140 @@ in
         Torrents are downloaded to ${homeDir}/${downloadsDir} by default and are
```
[the remark this answers is not in the document]
Comment (ju1m) [reply nixos/modules/services/torrent/transmission.nix]: I've changed to use `<xref linkend="opt-services.transmission.home"/>`

**Comment by ju1m** [comment]
> In addition to the things below, the descriptions of options `downloadDirPermissions` and `home` can be updated to reference the new options.
Done.

> Also as another improvement to this module: There's `homeDir` defined in a `let in` at the top which can be removed (its usages need to be fixed though).
Done.

**Review by picnoir — APPROVED** [review]
LGTM.

I don't see any obvious migration pitfall here apart from what's already been addressed.

Tested on my personal deployment.

I'll leave it there for 7 days before merging to let @Infinisil perform a second pass if he feels like it.

**Comment by NinjaTrappeur** [comment]
I'd be nice to have some VM tests for this module.

**Comment by ju1m** [comment]
- [X] Rebase after the merge of https://github.com/NixOS/nixpkgs/pull/139128

**Pull request merged**